extensions/src/main/java/org/pentaho/platform/plugin/services/metadata/PentahoMetadataDomainRepository.java

```diff
@@ -406,29 +412,35 @@ private boolean hasAccessFor( RepositoryFile repositoryFile ) {
    * file and multiple .properties files.
    */
   public Map<String, InputStream> getDomainFilesData( final String domainId ) {
-    Set<RepositoryFile> metadataFiles;
     lock.readLock().lock();
     try {
-      metadataFiles = metadataMapping.getFiles( domainId );
+      if ( !metadataMapping.getDomainIds().contains( domainId ) ) {
+        return emptyMap();
+      }
+      Map<String, InputStream> localeFiles =
+        ofNullable( metadataMapping.getLocaleFiles( domainId ) ).orElse( emptyMap() ).entrySet().stream()
+          .collect(
+            toMap( entry -> "messages_" + entry.getKey() + ".properties",
+              item -> getRepositoryFileInputStream( item.getValue() ) ) );
+      //    ^ keys of map are locale file names, e.g. "messages_fr_FR.properties"
+      return ImmutableMap.<String, InputStream>builder()
+        .putAll( localeFiles )
+        .put( domainId, getRepositoryFileInputStream( metadataMapping.getDomainFile( domainId ) ) )
+        .build();
     } finally {
       lock.readLock().unlock();
     }
+  }
 
-    Map<String, InputStream> values = new HashMap<String, InputStream>( metadataFiles.size() );
-    for ( RepositoryFile repoFile : metadataFiles ) {
-      RepositoryFileInputStream is;
-      try {
-        is = new RepositoryFileInputStream( repoFile );
-      } catch ( Exception e ) {
-        return null; // This pretty much ensures an exception will be thrown later and passed to the client
-      }
-      String fileName = repoFile.getName().endsWith( ".properties" ) ? repoFile.getName()
-        : domainId + ( domainId.endsWith( ".xmi" ) ? "" : ".xmi" );
```

Comment:
seems we've lost this part of logic related domainId ending in .xml.  do we still need it?

Reply:
From what I could see, domainId does consistently end with .xmi.

But you're right--  better safe than sorry.  I'll add a check.